### DISCUSSION TITLE
Option to convert empty environment variables to default values

### DISCUSSION THREAD

**Discussion by AndBondStyle** [pydantic/pydantic#3796] [Feature Request]
I think it would be useful to have an option in [BaseSettings.Config](https://github.com/samuelcolvin/pydantic/blob/be246701c504824f4bfc291209ac057ff7df0bc2/pydantic/env_settings.py#L79) that enables silent conversion of empty environment variables to default values. An example below is crashing because JSON can't parse empty string as valid object. Instead, I want it to just use provided default value.

```
from pydantic import BaseSettings
import os

class Settings(BaseSettings):
    test: dict = {"default": 123}

os.environ["TEST"] = ""
print(Settings())
```

My initial use case for this comes from docker-compose. I like to list all environment variables in `docker-compose.yml` using [variable substitution syntax](https://docs.docker.com/compose/environment-variables/):

```
environment:
  - "REQUIRED_VAR=${REQUIRED_VAR:?}"
  - "OPTIONAL_VAR=${OPTIONAL_VAR}"
```

This allows me to first define all variables in `.env` file, and than use them in command like `docker compose --env-file .env up`. The problem is, if I don't set an `OPTIONAL_VAR` in `.env` file, docker-compose doesn't skip it, but instead passes an empty string to container environment. In this specific case, it would be very helpful to just interpret this variable as "not set", and use the default value.

**Comment by JLHasson** [comment]
I'd also be interested in this feature. I'd argue that an expected behavior of unsetting an environment variable is for the system to take on the default, i.e. it is fairly common to do something like
```
FOO= ./bar
```
if `$FOO` is set to something you don't want passed to the script `bar` in your shell. In this case you would expect the script to take on the default value of `$FOO`.

**Comment by hramezani** [comment]
Just merge a [PR](https://github.com/pydantic/pydantic-settings/pull/198) which probably addresses this problem.

The `env_ignore_empty` flag will be available in next `pydantic-settings` release.
